# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../clang/naked-non-asm.patch
# Copyright (C) 2019 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- cfe-9.0.0.src/include/clang/Basic/DiagnosticSemaKinds.td.vanilla	2019-10-03 17:16:15.368000000 +0200
+++ cfe-9.0.0.src/include/clang/Basic/DiagnosticSemaKinds.td	2019-10-03 17:17:31.392000000 +0200
@@ -8669,7 +8669,7 @@
 def err_filter_expression_integral : Error<
   "filter expression type should be an integral value not %0">;
 
-def err_non_asm_stmt_in_naked_function : Error<
+def warn_non_asm_stmt_in_naked_function : Warning<
   "non-ASM statement in naked function is not supported">;
 def err_asm_naked_this_ref : Error<
   "'this' pointer references not allowed in naked functions">;
--- cfe-9.0.0.src/lib/Sema/SemaDecl.cpp.vanilla	2019-10-03 16:35:25.348000000 +0200
+++ cfe-9.0.0.src/lib/Sema/SemaDecl.cpp	2019-10-03 17:18:06.716000000 +0200
@@ -13859,10 +13859,10 @@
         if (RegisterVariables)
           continue;
         if (!isa<AsmStmt>(S) && !isa<NullStmt>(S)) {
-          Diag(S->getBeginLoc(), diag::err_non_asm_stmt_in_naked_function);
+          Diag(S->getBeginLoc(), diag::warn_non_asm_stmt_in_naked_function);
           Diag(FD->getAttr<NakedAttr>()->getLocation(), diag::note_attribute);
-          FD->setInvalidDecl();
-          break;
+          //FD->setInvalidDecl();
+          //break;
         }
       }
     }
